console_conf/controllers/identity.py

```diff
@@ -50,24 +52,31 @@ def get_core_version():
     return version
 
 
-def get_device_owner():
-    """ Check if device is owned """
+def get_managed():
+    """ Check if device is managed """
+    con = SnapdConnection('', '/run/snapd.socket')
+    return con.get('v2/system-info').json()['result']['managed']
+
 
-    # TODO: use proper snap APIs.
+def get_realname(username):
     try:
-        extrausers_fp = open('/var/lib/extrausers/passwd', 'r')
-    except FileNotFoundError:
-        return None
-    with extrausers_fp:
-        passwd_line = extrausers_fp.readline()
-        if passwd_line and len(passwd_line) > 0:
-            passwd = passwd_line.split(':')
-            result = {
-                'realname': passwd[4].split(',')[0],
-                'username': passwd[0],
-                'homedir': passwd[5],
+        info = pwd.getpwnam(username)
+    except KeyError:
+        return ''
+    return info.pw_gecos.split(',', 1)[0]
+
+
+def get_device_owner():
+    """ Get device owner, if any """
+    con = SnapdConnection('', '/run/snapd.socket')
+    for user in con.get('v2/users').json()['result']:
+        user = user['email'].split('@')[0]
+        if os.path.isdir('/home/' + user):
```

Comment:
this is wrong, for example for my user, I have this:

```json
{
    "result": [
        {
            "email": "first-name.last-name@canonical.com",
            "id": 1,
            "username": "lp-username"
        }
    ],
    "status": "OK",
    "status-code": 200,
    "type": "sync"
}
```

the `username` key should be what we check for in `/home`, because with the above user created on UC18, I will have `/home/lp-username`, not `/home/first-name.last-name`.

Reply:
(and my apologies if the LP comment wasn't specific enough, hopefully the above JSON output is helpful)